### ISSUE TITLE
Quiz features incomplete

### ISSUE THREAD

**Issue by ilebedev** [pwnall/igor#191]
Two big issues:

- [ ]  A: Resource files are available even if students haven't checked in.
- [ ]  B: Students can submit before they are checked in, and analysis printouts leak quiz secrets.

Quiz is tomorrow morning :|
We'll come up with  a workaround for `A`, but B is a big problem :(

**Comment by ilebedev** [comment]
In case it saves you any time at all, I did a half hour's worth of digging:

`A` and `B` seem fixable by adding to `released?` in the assignment model (i don't think this will interfere with `scheduled?`, which predicates students being able to check in)

`models/assignment.rb:115`
Add some logic such that the exam is only released for a student if the student is **checked in** and **confirmed** and **their exam session is currently in progress**.

Best!

**Comment by pwnall** [comment]
I agree with most of what you said. We plan to leave the exam materials accessible after the exam session end time, but remove the ability to submit solutions.

Once a student downloads files, the files are forever saved on his/her computer. Removing the download ability would give a false sense of security, whereas leaving the downloads available makes it clear to the system's users that other measures are required if preventing unauthorized distribution of the exam materials is desired.

Does this sound reasonable?

**Comment by ilebedev** [comment]
I agree!

Exam materials should then be available for a given student if (s)he is checked in, confirmed, and their exam session start time is in the past.

Ya?
